src/ng-pattern-restrict.js

```diff
@@ -217,19 +217,11 @@ angular.module('ngPatternRestrict', [])
             tryParseRegex(entryRegex);
           }
 
-          function notThrows(testFn, shouldReturnTruthy) {
-          	try {
-          		return testFn() || !shouldReturnTruthy;
-          	} catch (e) {
-          		return false;
-          	}
-          }
-
           function detectGetCaretPositionMethods() {
             var input = iElement[0];
-            if (notThrows(function () { return input.selectionStart; })) {
+            if (typeof(input.selectionStart) === 'function') {
```

Comment:
This should be a number: https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement